monitor/monitor.go

```diff
@@ -11,74 +13,37 @@ import (
 	ctClient "github.com/google/certificate-transparency-go/client"
 	"github.com/google/certificate-transparency-go/jsonclient"
 	"github.com/jmhodges/clock"
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promauto"
 )
 
-// monitorStat is a struct collecting up various prometheus metrics a monitor
-// will export/track.
-type monitorStats struct {
-	sthTimestamp *prometheus.GaugeVec
-	sthAge       *prometheus.GaugeVec
-	sthFailures  *prometheus.CounterVec
-	sthLatency   *prometheus.HistogramVec
-}
-
-const (
-	// sthTimeout controls how long should each STH fetch wait before timing out
-	sthTimeout = time.Second * 15
-)
-
-var (
-	// internetFacingBuckets are histogram buckets suitable for measuring
-	// latencies that involve traversing the public internet.
-	internetFacingBuckets               = []float64{.1, .25, .5, 1, 2.5, 5, 7.5, 10, 15, 30, 45}
-	stats                 *monitorStats = &monitorStats{
-		sthTimestamp: promauto.NewGaugeVec(prometheus.GaugeOpts{
-			Name: "sth_timestamp",
-			Help: "Timestamp of observed CT log signed tree head (STH)",
-		}, []string{"uri"}),
-		sthAge: promauto.NewGaugeVec(prometheus.GaugeOpts{
-			Name: "sth_age",
-			Help: "Elapsed time since observed CT log signed tree head (STH) timestamp",
-		}, []string{"uri"}),
-		sthFailures: promauto.NewCounterVec(prometheus.CounterOpts{
-			Name: "sth_failures",
-			Help: "Count of failures fetching CT log signed tree head (STH)",
-		}, []string{"uri"}),
-		sthLatency: promauto.NewHistogramVec(prometheus.HistogramOpts{
-			Name:    "sth_latency",
-			Help:    "Latency observing CT log signed tree head (STH)",
-			Buckets: internetFacingBuckets,
-		}, []string{"uri"}),
-	}
-)
+// internetFacingBuckets are histogram buckets suitable for measuring
+// latencies that involve traversing the public internet.
+var internetFacingBuckets = []float64{.1, .25, .5, 1, 2.5, 5, 7.5, 10, 15, 30, 45}
 
 // monitorCTClient is an interface that specifies the ctClient.LogClient
 // functions that the monitor package uses. This interface allows for easy
 // shimming of client methods with mock implementations for unit testing.
 type monitorCTClient interface {
 	GetSTH(context.Context) (*ct.SignedTreeHead, error)
+	AddChain(context.Context, []ct.ASN1Cert) (*ct.SignedCertificateTimestamp, error)
```

Comment:
I don't think this is necessary to change now, but I wanted to point out that you could make two interfaces: a `chainAdder` that the `submitter` uses, and a `sthGetter` that the `fetcher` uses. This would allow you to define narrower mocks for the `submitter` tests vs the `fetcher` tests. However, it's a minor optimization, and the test are already written, so let's not refactor this bit right now: just a possible optimization for similar work in the future.

Reply:
Good idea. I'll keep that in mind, thanks!